.github/workflows/deployment.yml

```diff
@@ -1,28 +1,49 @@
 name: Deployment
 
 on:
-  # Uncomment once we are ready to deploy 
-  #push:
-  #  branches:
-  #    - main
   workflow_dispatch:
+    # Uncomment once we are ready to deploy 
+    #push:
+    #  branches:
+    #    - main
+  pull_request:
```

Comment:
What would be the motivation for running this on every pull request rather than on pushes to main? 
Wouldn't this just be testing the configuration of the main branch and not the changes being introduced?
If we want to keep it as being triggered by pull requests can we only trigger it when files in src/website and .devcontainer are modified?

Reply:
From my understanding, running this on every PR would ensure that changes in the PR would not break the deployment before they are merged into the main branch, which would also provide another layer of testing to catch issues early. What do you think?

Also, good point. If we do want to keep it as being triggered by PR's, then it should only trigger when files in src/website and .devcontainer are modified. I will fix that

Reply:
Yeah sounds like a great idea to test out how changes deploy before actually deploying to the real site.  

~~The main branch will be checked out on the VM, and not the development branch~~
I'm sorry that's totally wrong ^

Would the VM that this workflow deploys to be like a pre-production environment and not the actual web host?

If so, then I think keeping it as triggered by PRs makes sense, and maybe we can keep the push trigger as well and deploy to either the pre-production VM or the actual website host VM depending on the trigger event.

What do you think?

Reply:
Your idea sounds great, since the VM that the workflow deploys will not be the actual web host.